app/main.js

```diff
@@ -1,109 +1,110 @@
 const fs = require("fs-extra");
 const settings = require("electron-settings");
+const path = require("path");
+const ipc = require("electron").ipcRenderer;
 
-let lastDone = "up"; // imporant var used around main.js
-
-// eslint-disable-next-line no-unused-vars
-function sayHello() {
-	console.log("Hello World!");
-}
-
-// Runs on file init, creates nessacary values for the file
-let homeFolder = "";
-let awayFolder = "";
-settings.get("paths.home").then((val) => {
-	homeFolder = val;
-	console.log(`awayFolder: ${val}`);
-});
-settings.get("paths.away").then((val) => {
-	awayFolder = val;
-	console.log(`homeFolder: ${val}`);
-});
-
-const filesList = [];
-fs.readdir(awayFolder, (err, files) => {
-	files.forEach((file) => {
-		console.log(file);
-		filesList.push(file);
+let ready = false;
+let filelocation = "up";
+const debug = true;
+function init() {
+	return new Promise((resolve, reject) => {
+		if (settings.has("path.home") && settings.has("path.away")) {
+			fs.readdir(settings.get("path.away"), (err, files) => {
+				if (err) {
+					reject(err);
+					return;
+				}
+				resolve(files);
+			});
+			ready = true;
+		}
 	});
+}
+let files;
+init().then(res => {
+	files = res;
+}).catch(err => {
+	console.error(err);
 });
-// End file init
-
-
 // eslint-disable-next-line no-unused-vars
 function down() {
-	if (lastDone == "down") {
-		notifyUser("The Files Are Currently Down On The Local Machine.", "blue", 2500);
-	} else {
-		lastDone = "down";
-		filesList.forEach((file) => {
-			fs.move(awayFolder + file, homeFolder + file, (err) => {
-				if (err != null) {
-            // the variable is defined
-					console.log(`error on down function file: ${file}`);
-          // printing what file this failed on
-					console.log(err);
-          // printing the error code
-					notifyUser("Down function has failed", "red");
-          // notifying the frontend that something has gone wrong, oops
-				} else {
-					console.log(`down function completed succesfully on file: ${file}`);
-					notifyUser("The File Have Succesfully Been Moved Down To The Local Machine!", "grey", 7000);
-				}
+	if (ready){
+		if (filelocation == "down") {
+			notifyUser("The Virtual Machine is currently down on the local machine!", "lightblue", 3000);
+		} else {
+			filelocation = "down";
+			let ifError = false;
+			files.forEach((file) => {
```

Comment:
Looks like up and down do the same thing minus some values, can we abstract this into a method and use it for both?